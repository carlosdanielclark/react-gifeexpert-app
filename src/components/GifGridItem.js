import React from 'react';
import '../stylesheet/GifGridItem.css';
import "animate.css";
import PropTypes from 'prop-types'; // ES6

const GifGridItem = ({ title, url })=>{
  return (
    <div className="animate__animated animate__bounce">
      <img src={url} alt={title} />
      <p> {title} </p>
    </div>
  );
}

export default GifGridItem;

GifGridItem.propTypes = {
  // You can declare that a prop is a specific JS primitive. By default, these
  // are all optional.
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,

}