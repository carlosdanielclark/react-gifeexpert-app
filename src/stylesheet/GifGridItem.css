.card{
  width: 170px;
  height: 210px;
  border: 1px solid #ddd;
  border-radius: 4px;
  margin: 3px;
}

img {
  width: 100%;
  height: 140px;
  border: 1px solid #ddd;
  border-radius: 4px;
}