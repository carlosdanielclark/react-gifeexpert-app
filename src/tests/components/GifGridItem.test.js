import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'; 
import GifGridItem from '../../components/GifGridItem'

describe('Testing component "GifGridItem.js"', ()=>{
  //(1) Inicializacion
  const title = "Hello Carlos";
  const url = "http://localhost:3000/algo.jpg";  

  test('Take Snapshot of the "title"', () => {

  //(2) Estimulo
  const wrapper = shallow( <GifGridItem 
    title={ title } 
    url = { url }
    /> );
  //(3) Observacion
  expect( wrapper ).toMatchSnapshot();
  });
  
});
  