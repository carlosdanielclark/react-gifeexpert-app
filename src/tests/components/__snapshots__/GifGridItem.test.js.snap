// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component "GifGridItem.js" Take Snapshot of the "title" 1`] = `
<div
  className="animate__animated animate__bounce"
>
  <img
    alt="Hello Carlos"
    src="http://localhost:3000/algo.jpg"
  />
  <p>
     
    Hello Carlos
     
  </p>
</div>
`;
